loadbalancer/algorithm_test.go

```diff
@@ -292,7 +292,16 @@ func TestApply(t *testing.T) {
 func TestConsistentHashSearch(t *testing.T) {
 	apply := func(key string, endpoints []string) string {
 		ch := newConsistentHash(endpoints).(*consistentHash)
-		return endpoints[ch.search(key, noSkippedEndpoints)]
+		ctx := &routing.LBContext{Route: &routing.Route{LBEndpoints: make([]routing.LBEndpoint, 0, len(endpoints))}}
+		for _, ep := range endpoints {
+			scheme, host, err := net.SchemeHost(ep)
+			if err != nil {
+				t.Fatal(err)
+			}
+			ctx.LBEndpoints = append(ctx.LBEndpoints, routing.LBEndpoint{Host: host, Scheme: scheme})
+			ctx.Route.LBEndpoints = append(ctx.Route.LBEndpoints, routing.LBEndpoint{Host: host, Scheme: scheme})
+		}
```

Comment:
Please extract this to helper. Also why do we do it manually, shouldn't there be a postprocessor for that?

Reply:
Yes, it's possible for postprocessor to do this. I think helper is `postprocessor.Do` in this case.